src/Raven.Server/Web/ServerRequestHandler.cs

```diff
@@ -0,0 +1,16 @@
+﻿using Raven.Client;
+
+namespace Raven.Server.Web
+{
+    public abstract class ServerRequestHandler : RequestHandler
+    {
+        public override void Init(RequestHandlerContext context)
+        {
+            base.Init(context);
+
+            var topologyEtag = GetLongFromHeaders(Constants.Headers.TopologyEtag);
+            if (topologyEtag.HasValue && Server.ServerStore.HasTopologyChanged(topologyEtag.Value))
+                context.HttpContext.Response.Headers[Constants.Headers.RefreshTopology] = "true";
```

Comment:
why is this required?

Reply:
It's for the `ClusterRequestExecutor` in the client. So we can update its topology if needed.

Reply:
I think it would be better to store this in memory in OnTopologyChangeInternal. This would avoid us opening transaction for each request and we are already doing that for DatabaseRequestHandler.

Reply:
done